src/arc.rs

```diff
@@ -39,6 +39,18 @@ pub(crate) struct ArcInner<T: ?Sized> {
 unsafe impl<T: ?Sized + Sync + Send> Send for ArcInner<T> {}
 unsafe impl<T: ?Sized + Sync + Send> Sync for ArcInner<T> {}
 
+impl<T: ?Sized> ArcInner<T> {
+    /// Safety: value must be a valid initialized pointer.
```

Comment:
nit: please document what this function does, and also document that it is a safety-usable invariant